go/carmen/memory_footprint.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package carmen
+
+import "github.com/Fantom-foundation/Carmen/go/common"
+
+type MemoryFootprint interface {
+	// Total provides the amount of bytes consumed by the database structure including all its subcomponents
+	Total() uint64
+
+	// String allow memory footprints to be used in format strings.
+	// It returns a breakdown of all MemoryFootprints within the tree.
+	String() string
+}
+
+func newMemoryFootprint(fp *common.MemoryFootprint) MemoryFootprint {
+	return &memoryFootprint{
+		fp: fp,
+	}
+}
+
+func NewMemoryFootprintFromValue(value uint64) MemoryFootprint {
```

Comment:
Is this method needed? If yes, I guess  it should be private anyway, we only want provide the footprint to users, but they do not need to create their own instances.